src/components/PeriodDimension/__tests__/FixedPeriodSingleSelect.spec.js

```diff
@@ -1,29 +1,14 @@
-import { shallow } from 'enzyme'
+import { render } from '@testing-library/react'
 import React from 'react'
 import FixedPeriodSelect from '../FixedPeriodSelect.js'
 
-describe('The Fixed Period Single Select component', () => {
-    let props
-    let shallowFixedPeriodSelect
-
-    const getWrapper = () => {
-        if (!shallowFixedPeriodSelect) {
-            shallowFixedPeriodSelect = shallow(<FixedPeriodSelect {...props} />)
-        }
-        return shallowFixedPeriodSelect
+test('FixedPeriodSingleSelect matches the snapshot', () => {
+    const props = {
+        value: '201405',
+        onChange: () => {},
     }
 
-    beforeEach(() => {
-        props = {
-            value: '201405',
-            onChange: () => {},
-        }
-        shallowFixedPeriodSelect = undefined
-    })
-
-    it('matches the snapshot', () => {
-        const wrapper = getWrapper()
+    const { container } = render(<FixedPeriodSelect {...props} />)
 
-        expect(wrapper).toMatchSnapshot()
-    })
+    expect(container).toMatchSnapshot()
```

Comment:
Same snapshot problem. Here I would consider creating mocks for  SingleSelectField, InputField, SingleSelectOption from ui. See here for an example: 

https://github.com/dhis2/maps-app/blob/master/src/components/layers/toolbar/__tests__/LayerToolbar.spec.js#L6

Reply:
Refactored this one to not rely on snapshot and without mocking.
I just look for some text in the rendered component and check that the selected values are correct.